File: app/src/main/java/ua/te/hackathon/smartcity2015/ui/main/MainPresenter.java
package ua.te.hackathon.smartcity2015.ui.main;

import android.support.annotation.Nullable;

import ua.te.hackathon.smartcity2015.ui.Presenter;

/**
 * @author victor
 * @since 2016-02-13
 */
public class MainPresenter implements Presenter<MainView> {
  @Nullable
  private MainView view;

  @Override
  public void attachView(MainView view) {
    this.view = view;
  }

  @Override
  public void detachView() {
    this.view = null;
  }

  @Override
  public void onDestroy() {

  }
}

File: app/src/main/java/ua/te/hackathon/smartcity2015/ui/main/MainPresenter.kt
package ua.te.hackathon.smartcity2015.ui.main

import android.content.Context

import ua.te.hackathon.smartcity2015.ui.base.mvp.Presenter

/**
 * @author victor
 * *
 * @since 2016-02-13
 */
class MainPresenter(private val appContext: Context) : Presenter<MainView> {
  private var view: MainView? = null

  override fun attachView(view: MainView) {
    this.view = view
  }

  override fun detachView() {
    this.view = null
  }

  override fun onDestroy() {
    detachView()
  }
}
